### PR TITLE
immich: 1.119.1 -> 1.120.1

### PR DESCRIPTION
**Pull request by dotlambda**
Changelog:
https://github.com/immich-app/immich/releases/tag/v1.120.0
https://github.com/immich-app/immich/releases/tag/v1.120.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Scrumplex** [comment]
https://github.com/immich-app/immich/pull/13773 added a regular database dump/backup job that utilized `pg_dumpall` and `gzip`. I assume we need to add these to `PATH` using our existing wrapper.

**Comment by dotlambda** [comment]
Is there a way to start the backup manually so we can run it in the test?

**Comment by Scrumplex** [comment]
> Is there a way to start the backup manually so we can run it in the test?

I think we should be able to trigger it using the following curl:

```console
$ curl -X PUT --json '{"command": "start"}' "$baseUrl/api/jobs/backupDatabase"
```

And check for completion

```console
$ curl "$baseUrl/api/jobs" | jq '.backupDatabase.jobCounts.completed > 0'
```

**Comment by teto** [comment]
dope ! I thought of suggesting a nixos module to backup immich this week (on the discussed nextcloud model) but this is too specific to be reusable properly. It was simpler to use the restic and postgresqlBackup module already. I might have to disable the latter if immich already does it !

**Comment by dotlambda** [comment]
Sounds like we're not supposed to use the built-in database backups: https://github.com/immich-app/immich/issues/13971#issuecomment-2461128994

File: nixos/modules/services/web-apps/immich.nix [hunk 1]
```diff
@@ -116,7 +116,7 @@ in
       description = ''
         Configuration for Immich.
         See <https://immich.app/docs/install/config-file/> or navigate to
-        <https://your-immich-domain/admin/system-settings> for
+        <https://my.immich.app/admin/system-settings> for
```
Comment (Titaniumtown) [inline nixos/modules/services/web-apps/immich.nix:119]: why this change? should we be linking to their site? I think the placeholder is fine.

**Review by Titaniumtown — CHANGES_REQUESTED** [review]

File: nixos/modules/services/web-apps/immich.nix (hunk 1 above)
Comment (dotlambda) [reply nixos/modules/services/web-apps/immich.nix:119]: The whole purpose of https://my.immich.app/ is to redirect to your Immich instance.

Comment (Titaniumtown) [reply nixos/modules/services/web-apps/immich.nix:119]: ok :shrug:

**Comment by emmertex** [comment]
```

Nov 08 09:40:35 nixserver0 immich[1229]: [Nest] 1229  - 11/08/2024, 9:40:35 AM   ERROR [Api:ErrorInterceptor~e6tivfna] Unknown error: Error: Perl must be installed. Please add perl to your $PATH and try again.
Nov 08 09:40:35 nixserver0 immich[1229]: Error: Perl must be installed. Please add perl to your $PATH and try again.
Nov 08 09:40:35 nixserver0 immich[1229]:     at /nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/node_modules/exiftool-vendored/dist/ExifTool.js:106:15
Nov 08 09:40:35 nixserver0 immich[1229]:     at async /nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/node_modules/exiftool-vendored/dist/ExifTool.js:153:17
Nov 08 09:40:35 nixserver0 immich[1229]:     at async f (/nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/node_modules/exiftool-vendored/dist/ExifTool.js:401:13)
Nov 08 09:40:35 nixserver0 immich[1229]:     at async g (/nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/node_modules/exiftool-vendored/dist/AsyncRetry.js:8:20)
Nov 08 09:40:35 nixserver0 immich[1229]:     at async ServerInfoRepository.getBuildVersions (/nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/dist/repositories/server-info.repository.js:78:23)
Nov 08 09:40:35 nixserver0 immich[1229]:     at async ServerService.getAboutInfo (/nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/dist/services/server.service.js:36:31)
Nov 08 09:40:40 nixserver0 immich[1229]: [Nest] 1229  - 11/08/2024, 9:40:40 AM   ERROR [Api:ErrorInterceptor~bwday2jk] Unknown error: Error: Perl must be installed. Please add perl to your $PATH and try again.
Nov 08 09:40:40 nixserver0 immich[1229]: Error: Perl must be installed. Please add perl to your $PATH and try again.
Nov 08 09:40:40 nixserver0 immich[1229]:     at /nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/node_modules/exiftool-vendored/dist/ExifTool.js:106:15
Nov 08 09:40:40 nixserver0 immich[1229]:     at async /nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/node_modules/exiftool-vendored/dist/ExifTool.js:153:17
Nov 08 09:40:40 nixserver0 immich[1229]:     at async f (/nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/node_modules/exiftool-vendored/dist/ExifTool.js:401:13)
Nov 08 09:40:40 nixserver0 immich[1229]:     at async g (/nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/node_modules/exiftool-vendored/dist/AsyncRetry.js:8:20)
Nov 08 09:40:40 nixserver0 immich[1229]:     at async ServerInfoRepository.getBuildVersions (/nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/dist/repositories/server-info.repository.js:78:23)
Nov 08 09:40:40 nixserver0 immich[1229]:     at async ServerService.getAboutInfo (/nix/store/nd9h4qd4a1v0fwp610xkb6baw56vrlry-immich-1.120.1/dist/services/server.service.js:36:31)
```

**Comment by dotlambda** [comment]
> Adding perl back in as removed [1df8bd3](https://github.com/NixOS/nixpkgs/commit/1df8bd30472fcbf43196bc7123b9fc4b30bec83e) results in a good bulid and all working.

I instead disabled the check whether perl is on the PATH.

**Comment by dotlambda** [comment]
I'll merge this today if nobody objects.
I also want to make people aware of https://github.com/immich-app/immich/discussions/14002.

File: nixos/tests/web-apps/immich.nix [hunk 2]
```diff
@@ -13,7 +13,10 @@ import ../make-test-python.nix (
           diskSize = 4096;
         };
 
-        environment.systemPackages = with pkgs; [ immich-cli ];
+        environment.systemPackages = with pkgs; [
+          immich-cli
+          jq
```
Comment (Scrumplex) [inline nixos/tests/web-apps/immich.nix:18]: not actually used?

File: nixos/modules/services/web-apps/immich.nix [hunk 3]
```diff
@@ -317,6 +317,11 @@ in
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
       inherit (cfg) environment;
+      path = [
+        # gzip and pg_dumpall are used by the backup service
+        pkgs.gzip
+        config.services.postgresql.package
+      ];
```
Comment (Scrumplex) [inline nixos/modules/services/web-apps/immich.nix:324]: Any reason why we don't add this to the already existing wrapper for immich? That would allow running it in contexts other than the service configured by the NixOS module.

Comment (dotlambda) [reply nixos/modules/services/web-apps/immich.nix:324]: I wasn't sure which PostgreSQL version to use. This is a way to use the "correct" version as long as the database is local.

**Review by Scrumplex — APPROVED** [review]
